src/components/annotations/defaults.js

```diff
@@ -65,7 +68,7 @@ function handleAnnotationDefaults(annIn, annOut, fullLayout) {
             // for now the arrow can only be on the same axis or specified as pixels
             // TODO: sometime it might be interesting to allow it to be on *any* axis
             // but that would require updates to drawing & autorange code and maybe more
-            if(aaxRef !== 'pixel' && aaxRef !== axRef) {
+            if(aaxRef !== 'pixel' && aaxRef !== axRefAxOnly) {
```

Comment:
At some point it may be nice to allow annotation arrow tail references to be axis domains too, but that doesn't need to be now. Until then, I think we need `aaxRef` to still match the complete `axRef` or revert to `'pixel'`.

Reply:
Re: your comment above about these attributes - you're right that specifying the annotation tail in anything other than pixels is fairly limited, ie it doesn't currently support `'paper'` at all. That happens in the `coerceRef` immediately above, where `'paper'` is not allowed as an extra, only `'pixel'`. Again, doesn't need to be added now - we just need to ensure that at the end of `supplyDefaults` we only have valid combinations of attributes given the currently implemented subset of features, regardless of what values the user feeds in.

Reply:
In the end I made it so that as long as `aaxRef === `axRef` (e.g., `xref === axref`), then it will draw both the annotation and the arrow using that coordinate system. However for all cases other than `pixel`, the coordinates of the arrow head are absolute, which is consistent with how it worked for data references (e.g., `xref === xref === 'x'`). The reason for this is that it made the logic actually weirder because we couldn't just do `if(aaxRef !== 'pixel' && aaxRef !== axRef)`, and I thought it would make users happy :) because it will be consistent with how it works for data coordinates.

Reply:
:+1: